arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptySet.kt

```diff
@@ -0,0 +1,47 @@
+package arrow.core
+
+public class NonEmptySet<out T>(
+  private val first: T,
+  rest: Set<T>,
+) : AbstractSet<T>() {
+
+  public val elements: Set<T>
+
+  init {
+      elements = setOf(first) + rest
+  }
```

Comment:
```suggestion
 public val elements: Set<T> = setOf(first) + rest
```

or even better, if we don't want to compute it until we need it

```suggestion
 public val elements: Set<T> by lazy { setOf(first) + rest }
```